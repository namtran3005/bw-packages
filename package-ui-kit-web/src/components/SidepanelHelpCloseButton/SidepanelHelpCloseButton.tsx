import * as React from 'react';
import styled from 'styled-components';
import { Spacing, Colors } from '@bitwala-cryptobank-squad/package-theme';
import { ButtonBase as MuiButtonBase } from '@material-ui/core';
import { ButtonBaseProps as MuiButtonBaseProps } from '@material-ui/core/ButtonBase';
import { ColorsType } from '@bitwala-cryptobank-squad/package-theme';

interface SidepanelHelpCloseButtonProps extends MuiButtonBaseProps {
  color?: ColorsType;
}

const SidepanelHelpCloseButton = styled(
  (props: SidepanelHelpCloseButtonProps) => (
    <MuiButtonBase {...props}>
      <div />
    </MuiButtonBase>
  )
).withConfig({ shouldForwardProp: (prop) => !['color'].includes(prop) })`
  &&& {
    width: ${Spacing.unit * 4}px;
    height: ${Spacing.unit * 4}px;
    color: ${Colors.get('white')};
    white-space: nowrap;
    border-radius: 0;
  }

  > div {
    width: ${Spacing.unit * 4}px;
    height: ${Spacing.unit * 4}px;

    &:before,
    &:after {
      content: '';
      position: absolute;
      width: 3px;
      height: ${Spacing.unit * 2 + 2}px;
      background: ${({ color }) =>
        color ? Colors.get(color) : Colors.get('white')};
      transform-origin: center;
      pointer-events: none;
      top: 25%;
      left: 47%;
    }

    &:before {
      transform: rotateZ(45deg);
    }

    &:after {
      transform: rotateZ(-45deg);
    }
  }
`;

export { SidepanelHelpCloseButton, SidepanelHelpCloseButtonProps };
