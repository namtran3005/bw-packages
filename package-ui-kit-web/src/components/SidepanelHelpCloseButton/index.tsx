export {
  SidepanelHelpCloseButton,
  SidepanelHelpCloseButtonProps,
} from './SidepanelHelpCloseButton';
