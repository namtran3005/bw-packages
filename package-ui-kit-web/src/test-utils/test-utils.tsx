import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
