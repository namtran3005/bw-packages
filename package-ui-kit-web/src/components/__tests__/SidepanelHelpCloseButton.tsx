import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { SidepanelHelpCloseButton } from '../SidepanelHelpCloseButton';

describe('SidepanelHelpCloseButton', () => {
  it('should render close button', () => {
    const { baseElement } = renderWithTestHarness(<SidepanelHelpCloseButton />);
    expect(baseElement).toMatchSnapshot();
  });
});
