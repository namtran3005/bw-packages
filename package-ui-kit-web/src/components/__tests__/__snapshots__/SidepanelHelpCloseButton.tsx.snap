// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidepanelHelpCloseButton should render close button 1`] = `
.c0.c0.c0 {
  width: 32px;
  height: 32px;
  color: #FFFFFF;
  white-space: nowrap;
  border-radius: 0;
}

.c0 > div {
  width: 32px;
  height: 32px;
}

.c0 > div:before,
.c0 > div:after {
  content: '';
  position: absolute;
  width: 3px;
  height: 18px;
  background: #FFFFFF;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  pointer-events: none;
  top: 25%;
  left: 47%;
}

.c0 > div:before {
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
}

.c0 > div:after {
  -webkit-transform: rotateZ(-45deg);
  -ms-transform: rotateZ(-45deg);
  transform: rotateZ(-45deg);
}

<body>
  <div>
    <button
      class="MuiButtonBase-root c0"
      tabindex="0"
      type="button"
    >
      <div />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
